var/spack/repos/builtin/packages/p7zip/package.py

```diff
@@ -0,0 +1,44 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class P7zip(MakefilePackage):
+    """A Unix port of the 7z file archiver"""
+
+    homepage = "http://p7zip.sourceforge.net"
+    url      = "https://downloads.sourceforge.net/project/p7zip/p7zip/16.02/p7zip_16.02_src_all.tar.bz2"
+
+    version('16.02', 'a0128d661cfe7cc8c121e73519c54fbf')
+
```

Comment:
Can you add a class attribute:
```python
builds_targets = ['all3']
```
and try removing `install` completely?

Reply:
And add `DEST_HOME=prefix` to `builds_targets` too.

Reply:
Not sure how to add DEST_HOME to the build targets. How do I access the prefix outside of a function's scope?

Reply:
Try:
```python
    @property
    def build_targets(self):
        return [ 'all3', 'DEST_HOME={0}'.format(prefix) ]
```

Reply:
Sometimes, the makefile needs to have `?=` instead of a simple `=` though.

Reply:
That's is a nice way to do it, but this makefile needs text substitution because it hard-codes a path.

Reply:
Neh, it has to go on `install_target` and you can drop the `edit` function:
```python
    @property
    def install_targets(self):
        return ['DEST_HOME={0}'.format(self.prefix), 'install']
```

Reply:
That worked. Thanks.